tests/models/vilt/test_modeling_vilt.py

```diff
@@ -530,7 +668,8 @@ def test_for_token_classification(self):
 
 # We will verify our results on an image of cute cats
 def prepare_img():
-    image = Image.open("./tests/fixtures/tests_samples/COCO/000000039769.png")
```

Comment:
@ydshieh maybe it's because of this line? Is `./tests/fixtures/tests_samples/COCO/000000039769.png` stored in the Docker image?

Reply:
It's in the repo, no? The CI clones the repo., so it should be there.

Thanks for the `_no_split_modules` not set part, now clear to me! So good for me to use seed for those tests.

Reply:
Great my bad then, reverted that change. However this seems to not fix the initial issue

Reply:
It's probably hardware / library version related, I saw somewhere that different PIL version can yield to different results

Reply:
Check here: https://github.com/huggingface/transformers/blob/bbbb453e5869a5fec4925b02f1265c9e6bfc3ebb/tests/models/vilt/test_modeling_vilt.py#L637

Reply:
By `the initial issue`,  you mean `ViltModelIntegrationTest::test_inference_natural_language_visual_reasoning`?
As it works on CI, I won't worry too much. I always try to Nvidia T4.

Reply:
I am not sure why you changed th line regarding image though.

Reply:
Yes I meant `ViltModelIntegrationTest::test_inference_natural_language_visual_reasoning` 
Just because I was running my tests not in the parent folder but from somewhere else, now that I am running the tests directly from the parent folder I prefer to keep the old changes

Reply:
Pillow versions indeed counts (in general) - we had some test failures before due to this.

Reply:
Great then I guess we can assume it is library version related! 💪 
Let me know if there is anything else which is unclear!

Reply:
Could we just set the seed, then call `super().test_cpu_offload()`?

Reply:
The only thing I am worried is that you need to set a seed before each inference step, so 2 times. Once before [the first forward pass](https://github.com/younesbelkada/transformers/blob/6a33f8a6388b24a9463a47a1822b5bff321bb3c7/tests/models/vilt/test_modeling_vilt.py#L543) and the second time before the [other forward passes ](https://github.com/younesbelkada/transformers/blob/6a33f8a6388b24a9463a47a1822b5bff321bb3c7/tests/models/vilt/test_modeling_vilt.py#L560) that is why I think that we cannot call `super().test_cpu_offload()` but have to define our own function

Reply:
Maybe there is a way to fix a seed without having to re-set it at each inference step

Reply:
Let me check

Reply:
I propose to add a context manager at 3cda114517680c8c215f82d90ff8a0a7b8844523
with that there is no need to redefine the whole function. We just need a decorator `set_reproducible` to be set on these functions.